scripts/delete_from_repository.py

```diff
@@ -0,0 +1,152 @@
+import argparse
+import logging
+import os
+import pprint
+import sys
+import time
+import typing
+from urllib.parse import urljoin
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+
+PROG_NAME = 'packages-repo-deleter'
+
+
+class RepositoryNotFound(ValueError):
+    pass
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser('packages-repo-deleter')
+    parser.add_argument('-r', '--repository-name', required=True, type=str)
+    parser.add_argument('-a', '--architecture', required=True, type=str)
+    parser.add_argument('-p', '--packages', nargs='+', required=True, type=str)
+    parser.add_argument('-v', '--verbose', action='store_true', default=False)
+    return parser.parse_args(args)
+
+
+def wait_for_task(base_url: str, task_href: str, auth: HTTPBasicAuth,
+                  logger: logging.Logger):
+    task_url = urljoin(base_url, task_href)
+    logger.info('Waiting for task %s to finish', task_href)
+    res = requests.get(task_url, auth=auth).json()
+    while res['state'] not in ('failed', 'completed'):
+        logger.debug('Still in progress...')
+        time.sleep(3)
+        res = requests.get(task_url, auth=auth).json()
+    logger.info('Task %s is finished', task_href)
+    return res
+
+
+def publish_repo(base_url: str, repo_href: str, auth: HTTPBasicAuth,
+                 logger: logging.Logger):
+    full_url = urljoin(base_url, '/pulp/api/v3/publications/rpm/rpm/')
+    task_href = requests.post(
+        full_url, json={'repository': repo_href}, auth=auth).json()['task']
+    return wait_for_task(base_url, task_href, auth, logger)
+
+
+def get_repository(base_url: str, repo_name: str, arch: str,
+                   auth: HTTPBasicAuth) -> typing.Optional[dict]:
+    full_url = urljoin(base_url, '/pulp/api/v3/repositories/rpm/rpm/')
+    params = {'name__contains': f'{repo_name}-{arch}'}
```

Comment:
why we can't search repo by exact name?
it's looks unsafe search repo without arch, we can get wrong repo

Reply:
Pulp does not have info on the repository architecture

Reply:
Fixed a search though, to ensure we pick up the correct repo